eventrecorder/recorder.go

```diff
@@ -0,0 +1,164 @@
+package eventrecorder
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-log/v2"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+var logger = log.Logger("lassie/event_recorder")
+
+type EventRecorder struct {
+	cfg    *config
+	server *http.Server
+	db     *pgxpool.Pool
+}
+
+func NewEventRecorder(ctx context.Context, opts ...option) (*EventRecorder, error) {
+	cfg, err := newConfig(
+		opts,
+		&config{
+			httpServerListenAddr:        "0.0.0.0:8080",
+			httpServerReadTimeout:       5 * time.Second,
+			httpServerReadHeaderTimeout: 5 * time.Second,
+			httpServerWriteTimeout:      5 * time.Second,
+			httpServerIdleTimeout:       10 * time.Second,
+			httpServerMaxHeaderBytes:    2048,
+		},
```

Comment:
Why repeat defaults here?

Reply:
Whoops. Artifact of an unfinished decision.